policy/modules/system/authlogin.fc

```diff
@@ -13,6 +13,7 @@ HOME_DIR/\.google_authenticator~		gen_context(system_u:object_r:auth_home_t,s0)
 /etc/gshadow.*		--	gen_context(system_u:object_r:shadow_t,s0)
 /etc/nshadow.*		--	gen_context(system_u:object_r:shadow_t,s0)
 /etc/shadow.*		--	gen_context(system_u:object_r:shadow_t,s0)
+/etc/tcb(/.*)?		--	gen_context(system_u:object_r:shadow_t,s0)
```

Comment:
The `--` specifier means the entry applies to a plain file only.

Reply:
So removing it would be the solution in this case?

Reply:
But I don't think we want to have directories labeled as `shadow_t` - it would likely lead to new denials, as there are no rules in the policy that would allow accessing `shadow_t` directories (except as part of the `file_type` attribute). So the current "label all regular files under /etc/tcb as `system_u:object_r:shadow_t:s0`" is exactly what we want here, IMO.

Reply:
Right. My point was to remove the '()' regexp part, but this is also reasonable as there are directories and symlinks in /etc/tcb, not just files. Thanks for pointing this out.

@besser82, is it expected a plain file which is not shadow-type appears in /etc/tcb? I understand there should only be `/etc/tcb/username/shadow` or `/etc/tcb/other/path/shadow` , and `username` can be a directory or a symlink somewhere else inside the tree. The regexp can be more strict then, like `/etc/tcb/[^/]+/.+` if it does not make it too hard.

Reply:
That's perfectly fine either.  I've changed that in the rebased commit.

Reply:
Using `/etc/tcb/(.*/)?shadow.*` should be fine then, I guess?